### PR TITLE
refactor(dot/rpc/modules) Add author unit tests, migrate old tests to integration tests

### PR DESCRIPTION
**Pull request by timwu20**
## Changes

- Moved existing tests to integration tests.  `-build=integration` is required when running these tests now to run the integration tests
- Added unit tests for `AuthorModule`

## Tests

```
go test ./dot/rpc/modules
```

## Issues

- #1472 
- #1026

### CODE REVIEW

File: dot/rpc/modules/author_integration_test.go [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+var testExt = common.MustHexToBytes("0x410284ffd43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d01f8efbe48487e57a22abf7e3acd491b7f3528a33a111b1298601554863d27eb129eaa4e718e1365414ff3d028b62bebc651194c6b5001e5c2839b982757e08a8c0000000600ff8eaf04151687736326c9fea17e25fc5287613693c912909cb226aa4794f26a480b00c465f14670")
+
+// invalid transaction (above tx, with last byte changed)
+//nolint
+var testInvalidExt = []byte{1, 212, 53, 147, 199, 21, 253, 211, 28, 97, 20, 26, 189, 4, 169, 159, 214, 130, 44, 133, 88, 133, 76, 205, 227, 154, 86, 132, 231, 165, 109, 162, 125, 142, 175, 4, 21, 22, 135, 115, 99, 38, 201, 254, 161, 126, 37, 252, 82, 135, 97, 54, 147, 201, 18, 144, 156, 178, 38, 170, 71, 148, 242, 106, 72, 69, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 216, 5, 113, 87, 87, 40, 221, 120, 247, 252, 137, 201, 74, 231, 222, 101, 85, 108, 102, 39, 31, 190, 210, 14, 215, 124, 19, 160, 180, 203, 54, 110, 167, 163, 149, 45, 12, 108, 80, 221, 65, 238, 57, 237, 199, 16, 10, 33, 185, 8, 244, 184, 243, 139, 5, 87, 252, 245, 24, 225, 37, 154, 163, 143}
```
Comment (noot) [inline dot/rpc/modules/author_integration_test.go:28]: in general we should try to get rid of these hard coded extrinsics and such and use the centrifuge library to generate them

File: dot/rpc/modules/author_test.go [hunk 2]
```diff
@@ -2,304 +2,359 @@ package modules
...
-	auth := NewAuthorModule(nil, nil, nil, txQueue)
+	// invalid transaction (above tx, with last byte changed)
+	//nolint
+	var testInvalidExt = []byte{1, 212, 53, 147, 199, 21, 253, 211, 28, 97, 20, 26, 189, 4, 169, 159, 214, 130, 44, 133, 88, 133, 76, 205, 227, 154, 86, 132, 231, 165, 109, 162, 125, 142, 175, 4, 21, 22, 135, 115, 99, 38, 201, 254, 161, 126, 37, 252, 82, 135, 97, 54, 147, 201, 18, 144, 156, 178, 38, 170, 71, 148, 242, 106, 72, 69, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 216, 5, 113, 87, 87, 40, 221, 120, 247, 252, 137, 201, 74, 231, 222, 101, 85, 108, 102, 39, 31, 190, 210, 14, 215, 124, 19, 160, 180, 203, 54, 110, 167, 163, 149, 45, 12, 108, 80, 221, 65, 238, 57, 237, 199, 16, 10, 33, 185, 8, 244, 184, 243, 139, 5, 87, 252, 245, 24, 225, 37, 154, 163, 143}
 
-	res := new(PendingExtrinsicsResponse)
-	err := auth.PendingExtrinsics(nil, nil, res)
-	if err != nil {
-		t.Fatal(err)
+	// ehr := ExtrinsicHashResponse(types.Extrinsic(testInvalidExt).Hash().String())
```
Comment (noot) [inline dot/rpc/modules/author_test.go:32]: remove?

**Review by noot — COMMENTED** [review]
how does `-build=integration` work? are the skipped tests not skipped if that's used?

**Review by EclesioMeloJunior — APPROVED** [review]

**Comment by timwu20** [comment]
> how does `-build=integration` work? are the skipped tests not skipped if that's used?

We will need to modify the CI to run `go test --tags=integration` to actually run these tests.

File: dot/rpc/modules/author_integration_test.go (hunk 1 above)
Comment (timwu20) [reply dot/rpc/modules/author_integration_test.go:28]: I want to move these tests as integration tests for `core.Service`.  I'll keep this in mind for the next PR.

**Review by noot — APPROVED** [review]

**Pull request merged**